[HUDI-590] Cut doc version for 0.5.1 and update site

bhasudha opened the pull request:
## *Tips*
- *Thank you very much for contributing to Apache Hudi.*
- *Please review https://hudi.apache.org/contributing.html before opening a pull request.*

## What is the purpose of the pull request

*(For example: This pull request adds quick-start document.)*

## Brief change log

*(for example:)*
  - *Modify AnnotationLocation checkstyle rule in checkstyle.xml*

## Verify this pull request

*(Please pick either of the following options)*

This pull request is a trivial rework / code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

*(example:)*

  - *Added integration tests for end-to-end.*
  - *Added HoodieClientWriteTest to verify the change.*
  - *Manually verified the change by running a job locally.*

## Committer checklist

 - [x] Has a corresponding JIRA in PR title & commit
 
 - [x] Commit message is descriptive of the change
 
 - [ ] CI is green

 - [x] Necessary doc changes done or have another open PR
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

bhasudha commented:
@leesf  @vinothchandar  Please note there are two commits in this PR for ease of review. I can squash it once after addressing review commands.

bhasudha commented:
@lamber-ken @leesf The cn docs are not uptodate. We might want to file a Jira and followup on these later.

leesf commented:
> @lamber-ken @leesf The cn docs are not uptodate. We might want to file a Jira and followup on these later.

sync. https://jira.apache.org/jira/projects/HUDI/issues/HUDI-666

vinothchandar commented:
LGTM as well.. 

+1 on updating the release guide with a pointer to the version cut on docs step

bhasudha commented:
@leesf @vc Thanks for the reviews. Updated the https://cwiki.apache.org/confluence/display/HUDI/Apache+Hudi+%28incubating%29+-+Release+Guide  with the section `Steps to cut doc version and update website` just after Finalizing release. 

Also I have squashed the commits into one. I can merge if you are okay with the changes.

leesf commented:
> @leesf @vc Thanks for the reviews. Updated the https://cwiki.apache.org/confluence/display/HUDI/Apache+Hudi+%28incubating%29+-+Release+Guide with the section `Steps to cut doc version and update website` as a last step in `Finalize the release`
> 
> Also I have squashed the commits into one. I can merge if you are okay with the changes.

LGTM, please merge.
